File: src/main/java/training/official/classic_puzzle_easy/TheDescent.java
package training.official.classic_puzzle_easy;

import java.util.Scanner;

class TheDescent {
	public static void main(String args[]) {
		Scanner in = new Scanner(System.in);

		while (true) {
			int target = 0;
			int targetH = 0;

			for (int i = 0; i < 8; i++) {
				int mountainH = in.nextInt();

				if (targetH < mountainH) {
					target = i;
					targetH = mountainH;
				}
			}

			System.out.println(target);
		}
	}
}

File: src/main/kotlin/training/official/classic_puzzle_easy/TheDescent.kt
//package training.official.classic_puzzle_easy

import java.util.*

fun main() {
	val scanner = Scanner(System.`in`)

	while (true) {
		(0 until 8)
			.map { TheDescent.Input(it, scanner.nextInt()) }
			.maxByOrNull { it.mountainHeight }
			?.also { println(it.index) }
	}
}

private class TheDescent { // only here for codingame
	data class Input(val index: Int, val mountainHeight: Int)
}
